integration_test/third_party_apps_data/applications/postgresql/metadata.yaml

```diff
@@ -31,31 +31,145 @@ minimum_supported_agent_version:
 supported_app_version: ["10.18+"]
 expected_metrics:
-  - type: workload.googleapis.com/postgresql.backends
+  - kind: GAUGE
+    labels:
+      database: .*
+    monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.backends
+    value_type: INT64
+    representative: true
+  - kind: CUMULATIVE
+    monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.bgwriter.buffers.allocated
+    value_type: INT64
+  - kind: CUMULATIVE
+    labels:
+      source: backend|backend_fsync|checkpoints|bgwriter
+    monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.bgwriter.buffers.writes
+    value_type: INT64
+  - kind: CUMULATIVE
+    labels:
+      type: requested|scheduled
+    monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.bgwriter.checkpoint.count
+    value_type: INT64
+  - kind: CUMULATIVE
+    labels:
+      type: sync|write
+    monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.bgwriter.duration
     value_type: INT64
-    kind: GAUGE
+  - kind: CUMULATIVE
     monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.bgwriter.maxwritten
+    value_type: INT64
+  - kind: CUMULATIVE
     labels:
       database: .*
-    representative: true
-  - type: workload.googleapis.com/postgresql.commits
+      table: .*
+      source: heap_read|heap_hit|idx_read|idx_hit|toast_read|toast_hit|tidx_read|tidx_hit
+    monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.blocks_read
+    value_type: INT64
+  - kind: CUMULATIVE
+    labels:
+      database: .*
+    monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.commits
     value_type: INT64
-    kind: CUMULATIVE
+  - kind: GAUGE
     monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.connection.max
+    value_type: INT64
+  - kind: GAUGE
+    monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.database.count
+    value_type: INT64
+  - kind: GAUGE
     labels:
       database: .*
-  - type: workload.googleapis.com/postgresql.db_size
+    monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.db_size
     value_type: INT64
-    kind: GAUGE
+  - kind: CUMULATIVE
+    labels:
+      database: .*
+      table: .*
+      index: .*
     monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.index.scans
+    value_type: INT64
+  - kind: GAUGE
     labels:
       database: .*
-  - type: workload.googleapis.com/postgresql.rollbacks
+      table: .*
+      index: .*
+    monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.index.size
     value_type: INT64
-    kind: CUMULATIVE
+  - kind: CUMULATIVE
+    labels:
+      database: .*
+      table: .*
+      operation: ins|upd|del|hot_upd
+    monitored_resource: gce_instance
+    type: workload.googleapis.com/postgresql.operations
```

Comment:
The scope [doc](https://docs.google.com/document/d/1M9r8CqZl2W9fYz8brzlWntEAqJUC1K43LHGvGkbpiPs/edit) has this as `workload.googleapis.com/postgresql.row.operations` with labels `operation=delete|fetch|hot_update|insert|return|update`.

Is this an intentional change to the scope doc? If so, can we send a suggested edit to the scope doc and have the approval double check it? If not, can we fix the translation so it matches the scope doc?

Reply:
This was intentional as this was the pre-existing metric `postgresql.operations` was not sure it was worth breaking users to make this name change. Will make a suggestion to scope doc.

Reply:
https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/third-party/postgresql#monitored-metrics

From the list of metrics above it doesn't seem like `postgresql.operations` was ever monitored (or atleast doesn't seem so from the public documentation and the metadata.yaml file)

This is interesting to me though because of: [b/246310954](http://b/246310954) | P2 | integration_test: test that no metric is reported for third party apps that don't have a corresponding metadata.yaml entry. 

Are we actually collecting more metrics quietly and not documenting? I looked at the old Postgres scope doc from last year and it says we will monitor the same as the legacy agent metrics: https://cloud.google.com/monitoring/api/metrics_agent#agent-postgresql

But that is a much larger list than what we claim to collect. We should double check what we're collecting and have the metadata and public docs reflect that.

Reply:
Hmm interesting... There were metrics added it appears from november 2021 into Otel https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/09d2b3eb52d6870ecbb46dbcc29dbff88294fe27

Also appears the docs (now replaced by metadata.yaml) were updated, but this was before the metadata.yaml seemed to have a strong foothold. https://github.com/GoogleCloudPlatform/ops-agent/pull/320/files

Metrics probably wouldn't have shown up because no user table was created. In the `exercise` I create tables and indexes to get these metrics to show up.

Reply:
![image](https://user-images.githubusercontent.com/18472685/189687774-4d180f14-8cbe-4a8f-9b92-059fb228e1fb.png)

I did a very unscientific test to see what was being collected now. Looking at the integration test VMs and what they collected a few months ago, it looks like only the publicly documented metrics were actually being collected. For all other metrics (added in this PR probably), they had no data. This implies to me that we actually only collected the publicly documented ones since the integration test does run the `enable` and `exercise` scripts. But can we verify this with a manual install and run?

I suspect you're right about these metrics being collected only in some circumstances and our docs and metadata files don't capture that scenario. If that is the case, this should be fixed